source/fe/fe_poly_tensor.cc

```diff
@@ -49,30 +49,29 @@ namespace internal
        * implements an algorithm that determines those DoFs that need this
        * sign change for a given cell.
        */
+      template <int spacedim>
       void
       get_face_sign_change_rt(const dealii::Triangulation<1>::cell_iterator &,
-                              const unsigned int,
-                              std::vector<double> &face_sign)
+                              const FiniteElement<1, spacedim> &,
+                              const std::vector<MappingType> &,
+                              std::vector<double> &)
       {
         // nothing to do in 1d
-        std::fill(face_sign.begin(), face_sign.end(), 1.0);
       }
 
 
 
+      //      template<int spacedim>
       void
       get_face_sign_change_rt(
         const dealii::Triangulation<2>::cell_iterator &cell,
-        const unsigned int                             dofs_per_face,
+        const FiniteElement<2, 2> &                    fe,
+        const std::vector<MappingType> &               mapping_type,
         std::vector<double> &                          face_sign)
       {
         const unsigned int dim      = 2;
         const unsigned int spacedim = 2;
 
-        // Default is no sign
-        // change. I.e. multiply by one.
-        std::fill(face_sign.begin(), face_sign.end(), 1.0);
```

Comment:
Shouldn't we still fill with ones by default? I don't understand why this is no longer needed.

Reply:
Lines 380, 961, 1595 each fill this with ones and then call `get_face_sign_change_rt` and `get_face_sign_change_nedelec` consecutively, so there's no need for the redundant fill with ones.

The idea is to stack possible sign changes for both rt and nedelec in the same vector. Filling with ones inside `get_face_sign_change_rt` or `get_face_sign_change_nedelec` could overwrite sign change information from the other function, and may not generalize as well should there be a need for another sign change function in the future.